src/utils/reqwest.rs

```diff
@@ -12,6 +15,21 @@ pub fn default_retry_policy() -> ExponentialBackoff {
     ExponentialBackoff::builder().build_with_max_retries(3)
 }
 
+fn auth_middleware() -> AuthenticationMiddleware {
+    if let Ok(auth_file) = std::env::var("RATTLER_AUTH_FILE") {
+        tracing::info!("Loading authentication from file: {:?}", auth_file);
+
+        let mut store = AuthenticationStorage::new();
+        store.add_backend(Arc::from(
+            authentication_storage::backends::file::FileStorage::new(PathBuf::from(&auth_file)),
```

Comment:
Can't this function fail? I expected this to provide a Result, what happens when the file is of incorrect format?

Reply:
It never fails currently. It would only fail later on if the format is incorrect (when reading / writing).

Reply:
And then it should log a warning.

Reply:
It actually doesn't and it is hard for the user to understand what it does. 
The error on borked or missing file:
```
➜ pixi global install pytorch -c https://prefix.dev/ruben-arts --auth-file ~/credentials.json
ERROR rattler_repodata_gateway::fetch: error=Request error: HTTP status client error (403 Forbidden) for url (https://prefix.dev/ruben-arts/linux-64/repodata.json)
  × failed to fetch repodata from channels
  ├─▶ Request error: HTTP status client error (403 Forbidden) for url (https://prefix.dev/ruben-arts/linux-64/repodata.json)
  ╰─▶ HTTP status client error (403 Forbidden) for url (https://prefix.dev/ruben-arts/linux-64/repodata.json)
```